hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/EntityCollectionPart.java

```diff
@@ -241,8 +242,12 @@ public String toString() {
 
 	@Override
 	public ForeignKeyDescriptor getForeignKeyDescriptor() {
-		// todo (6.0) : this will not strictly work - we'd want a new ForeignKeyDescriptor that points the other direction
-		return collectionDescriptor.getAttributeMapping().getKeyDescriptor();
+		if ( collectionDescriptor.isInverse() ) {
```

Comment:
I don't know exactly what the reason for this is, but it feels wrong to me that we could return the FK descriptor for the index part of the collection. If there is a problem with the direction of the foreign key, this should be handled by returning a different `ForeignKeyDescriptor.Nature` in the method below. At least this is how `ToOneAttributeMapping`s are handled.

IMO this should always return `return collectionDescriptor.getAttributeMapping().getKeyDescriptor( nature );`